fpdf/template.py

```diff
@@ -147,6 +147,29 @@ def _parse_multiline(s):
             return False
         return None
 
+    def parse_json(self, infile, encoding="utf-8"):
+        """
+        Load the template definition from a JSON file.
+        The data must be structured as an array of objects, with names and values exactly
+        equivalent to what would get supplied to load_elements(),
+
+        Arguments:
+
+            infile (string): The filename of the CSV file.
+
+            encoding (string): The character encoding of the file. Default is UTF-8.
+        """
+        with open(infile, encoding=encoding) as f:
+            data = json.load(f)
+            for d in data:
+                fgval = d.get("foreground")
+                if fgval and isinstance(fgval, str) and fgval.lower().startswith("0x"):
+                    d["foreground"] = int(fgval, 16)
+                bgval = d.get("background")
+                if bgval and isinstance(bgval, str) and bgval.lower().startswith("0x"):
+                    d["background"] = int(bgval, 16)
```

Comment:
Why not using `#rrggbb` hex colors instead?
There are already accepted in the `fpdf.html` & `fpdf.svg` modules.

I know that emplates already accept `0x`-prefixed hex colors,
but maybe this is the opportunity to improve the library consistency regarding colors handling,
and also allow `#rrggbb` hex colors there?

Reply:
Also, you could maybe call/reuse the `_parse_colorcode()` in this code block?

Reply:
I simply didn't think of that... :wink:
It's actually nicer than the explicit "0x..."

`_parse_colorcode()` isn't really suitable here, because JSON already returns "true" numerical values, while in CSV everyting is returned as a string that needs to be converted. The "#rrggbb" format makes this distinction even more explicit.